### PR TITLE
Removes check of CUDA runtime and linked libs from the backend

### PR DESCRIPTION
**Pull request by JanuszL**
- when DALI is imported backend module runs a check if the CUDA
  runtime and necessary libraries are available. If not it prints
  a warning. These checks initialize CUDA runtime what is causing
  trouble when the process is forked - for example
  torch.multiprocessing
- the checks are moved to check_cuda_runtime() and it is called
  just before the pipeline is constructed, not when DALI itself
  is imported

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

## Category:

**Bug fix** (*non-breaking change which fixes an issue*)

## Description:
- when DALI is imported backend module runs a check if the CUDA
  runtime and necessary libraries are available. If not it prints
  a warning. These checks initialize CUDA runtime what is causing
  trouble when the process is forked - for example
  torch.multiprocessing
- the checks are moved to check_cuda_runtime() and it is called
  just before the pipeline is constructed, not when DALI itself
  is imported

### Affected modules and functionalities:
- backend
- pipeline

### Key points relevant for the review:
- NA

### Tests
- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

**Review by mzient — APPROVED** [review]

File: dali/python/nvidia/dali/backend.py [hunk 1]
```diff
@@ -45,20 +45,29 @@ def deprecation_warning(what):
...
+        cuda_checked = True
+        if GetCudaVersion() == -1:
+            deprecation_warning("GPU is not available. Only CPU operators are available.")
 
-    if GetNppVersion() == -1:
-        deprecation_warning("Cannot access NPP library. Please check cuda installation and/or "
-                            "if is an appropriate wheel is installed.")
+        if GetCufftVersion() == -1:
+            deprecation_warning("Cannot access cuFFT library. Please check cuda installation and/or "
+                                "if is an appropriate wheel is installed.")
```
Comment (jantonguirao) [inline dali/python/nvidia/dali/backend.py:65]:
```suggestion
            deprecation_warning("Cannot access cuFFT library. Please check cuda installation and/or "
                                "if an appropriate wheel is installed.")
```

Comment (jantonguirao) [reply dali/python/nvidia/dali/backend.py:65]: same in the other two

**Review by jantonguirao — APPROVED** [review]

File: dali/python/nvidia/dali/backend.py (hunk 1 above)
Comment (JanuszL) [reply dali/python/nvidia/dali/backend.py]: Done

**Comment by JanuszL** [comment]
!build

**Pull request merged**